src/theory/theory_preprocessor.cpp

```diff
@@ -24,31 +26,76 @@ namespace CVC4 {
 namespace theory {
 
 TheoryPreprocessor::TheoryPreprocessor(TheoryEngine& engine,
-                                       RemoveTermFormulas& tfr)
+                                       RemoveTermFormulas& tfr,
+                                       ProofNodeManager* pnm)
     : d_engine(engine),
       d_logicInfo(engine.getLogicInfo()),
       d_ppCache(),
-      d_tfr(tfr)
+      d_tfr(tfr),
+      d_pfContext(),
+      d_tpg(pnm ? new TConvProofGenerator(
+                      pnm, &d_pfContext, TConvPolicy::FIXPOINT)
+                : nullptr),
+      d_lp(pnm ? new LazyCDProof(pnm, nullptr, &d_pfContext) : nullptr)
 {
+  if (isProofEnabled())
```

Comment:
This test looks at `d_tpg`, which is set above based on `pnm`. So better to just test on pnm? Moreover, the test below of pnm being not null should always suceed?